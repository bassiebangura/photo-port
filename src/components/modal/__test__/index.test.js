import React from "react";
import { cleanup, fireEvent, render, screen  } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Modal from "..";

afterEach(cleanup);

const mockCurrentPhoto = {
    name: "Grocery aisle",
    image: "commercial",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie",
    index: 0
};


const mockOnClose = jest.fn();

describe("Modal component", () => {
    // First Test
    it("renders", () => {
        render(<Modal currentPhoto={mockCurrentPhoto} onClose={mockOnClose} />);
    });
    // Second Test
    it("matches snapshot", () => {
        const { asFragment } = render(<Modal currentPhoto={mockCurrentPhoto} onClose={mockOnClose}></Modal>);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
    });  
})

describe("Click Event", () => {
    it("calls onClose handler", () => {
        render(<Modal currentPhoto={mockCurrentPhoto} onClose={mockOnClose}></Modal>);
        // Act: Simulate click event
        fireEvent.click(screen.getByText("Close Modal"));
        // Assert: Expected matcher
        expect(mockOnClose).toHaveBeenCalledTimes(1);
    });
})