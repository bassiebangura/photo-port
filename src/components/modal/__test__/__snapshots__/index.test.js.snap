// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="modalBackdrop"
    >
      <div
        class="modalContainer"
      >
        <h3
          class="modalTitle"
        >
          Photo Name
        </h3>
        <img
          alt="current category"
          src="0.jpg"
        />
        <p>
          Photo Description
        </p>
        <button>
          Close Modal
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
